src/org/gk/qualityCheck/HumanReactionsWithNonHumanPhysicalEntitiesWithoutDiseaseCheck.java

```diff
@@ -0,0 +1,100 @@
+package org.gk.qualityCheck;
+
+import org.gk.database.InstanceListPane;
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.GKSchemaClass;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+/**
+ * Flags all human Reactions that contain Non-only-Human PhysicalEntities that do not have a populated disease attribute.
+ * Valid PhysicalEntities are those that have a non-human species or that have a human species AND have a filled relatedSpecies attribute.
+ */
+public class HumanReactionsWithNonHumanPhysicalEntitiesWithoutDiseaseCheck extends AbstractQualityCheck {
+
+    @Override
+    public QAReport checkInCommand() throws Exception {
+        QAReport report = new QAReport();
+        MySQLAdaptor dba = (MySQLAdaptor) dataSource;
+        QACheckUtilities.setHumanSpeciesInst(dba);
+        QACheckUtilities.setSkipList(Files.readAllLines(Paths.get("QA_SkipList/Manually_Curated_NonHuman_Pathways.txt")));
+
+        // This QA Check is only performed on human ReactionlikeEvents that do not have any inferredFrom referrals.
+        for (GKInstance reaction : QACheckUtilities.findHumanReactionsNotUsedForManualInference(dba)) {
+            for (GKInstance reactionPE : QACheckUtilities.findAllPhysicalEntitiesInReaction(reaction)) {
+                // Valid PhysicalEntities include those that have a non-human species OR have a human species AND have a relatedSpecies,
+                // and that do not have a populated disease attribute.
+                if ((QACheckUtilities.hasNonHumanSpecies(reactionPE) || hasHumanSpeciesWithRelatedSpecies(reactionPE))
+                        && reactionPE.getAttributeValue(ReactomeJavaConstants.disease) == null) {
```

Comment:
Since the && has higher precedence, the last two conditions would get grouped to be the right-hand-side result of the ||.  Could the last two conditions be put in parentheses just to make it clear they are one unit?

Reply:
The first two conditions that are `||`d are in parentheses.

Reply:
Yea, its `(condition1 || condition2) && condition 3`

Reply:
Sorry! I missed the parens around the `||`.  Don't listen to me regarding this :).